### PR TITLE
Decode optimizations

### PR DESCRIPTION
**Pull request by omnivion**
### Information
- JIRA story QPPCT-413.

### Changes proposed in this PR:
- Stopped using reflection for namespace code

### Checklist 
- [X] All JUnit tests pass (`mvn clean verify`).
- [X] New unit tests written to cover new functionality.
- [X] Added and updated JavaDocs for non-test classes and methods.
- [X] No local design debt. Do you feel that something is "ugly" after your changes?
- [X] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: converter/src/main/java/gov/cms/qpp/conversion/decode/QppXmlDecoder.java [hunk 1]
```diff
@@ -59,15 +59,13 @@ public QppXmlDecoder(Context context) {
...
+		if (element == null) {
 			return DecodeResult.ERROR;
 		}
 
 		setNamespace(element, this);
 
-		DecodeResult decodeResult = decodeChildren(element, parentNode);
-
-		return (decodeResult != null) ? decodeResult : DecodeResult.TREE_CONTINUE;
+		return decodeChildren(element, parentNode);
```
Comment (halprin) [inline converter/src/main/java/gov/cms/qpp/conversion/decode/QppXmlDecoder.java]: We are losing the possibility of not returning `DecodeResult.TREE_CONTINUE` if `decodeChildren` returns `null`.

Comment (omnivion) [reply converter/src/main/java/gov/cms/qpp/conversion/decode/QppXmlDecoder.java]: All tests pass, so I'm assuming this should never happen & is therefore a pointless check. I can readd it if you guys want, but I don't see much point.

**Comment by FlexionSonarQube** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [XmlInputDecoder.java#L3](https://github.com/CMSgov/qpp-conversion-tool/blob/1b71b912fdcf06f5be01b068e71172cf56beca51/converter/src/main/java/gov/cms/qpp/conversion/decode/XmlInputDecoder.java#L3): Remove this unused import 'java.lang.reflect.Constructor'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [XmlInputDecoder.java#L4](https://github.com/CMSgov/qpp-conversion-tool/blob/1b71b912fdcf06f5be01b068e71172cf56beca51/converter/src/main/java/gov/cms/qpp/conversion/decode/XmlInputDecoder.java#L4): Remove this unused import 'java.lang.reflect.InvocationTargetException'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: converter/src/main/java/gov/cms/qpp/conversion/decode/XmlInputDecoder.java [hunk 2]
```diff
@@ -83,15 +83,10 @@ protected void decode(List<Element> elements, Node parent) {
...
-			constructor.setAccessible(true);
-			decoder.xpathNs = constructor.newInstance("ns", decoder.defaultNs.getURI());
-		} catch (NoSuchMethodException | InstantiationException | IllegalAccessException
-				| IllegalArgumentException | InvocationTargetException e) {
-			throw new IllegalArgumentException("Cannot construct special Xpath namespace", e);
-		}
+		// this handles the case where there is no URI for a default namespace (test)
+				String uri = decoder.defaultNs.getURI();
+				decoder.xpathNs = Strings.isNullOrEmpty(uri) ? Namespace.NO_NAMESPACE
+						: Namespace.getNamespace("ns", decoder.defaultNs.getURI());
```
Comment (clydet) [inline converter/src/main/java/gov/cms/qpp/conversion/decode/XmlInputDecoder.java]: Can you tidy the indentation here?

**Pull request merged**